packages/logger/README.md

```diff
@@ -1,4 +1,320 @@
+
+### [TypeScript](https://www.typescriptlang.org/) : Type System on ECMAScript 6
+
+1. Getting Started With TypeScript -- https://basarat.gitbooks.io/typescript/docs/getting-started.html 
+2. Declaration Files -- https://basarat.gitbooks.io/typescript/docs/types/ambient/d.ts.html
+3. Decorators -- https://www.typescriptlang.org/docs/handbook/decorators.html
+
+When you need to `npm install async` don't forget to `npm install -S "@types/async"
+
+### [OpenAPI](https://www.openapis.org/)
+
+1. https://editor.swagger.io
+
+Note (LoopBack Next specific): In a LoopBack Next app, typically you don't put the header portion of the api spec(see below) because LoopBack Next adds it when internalizing it.  The swagger editor will complain without the header portion.
+```
+{
+  "swagger": "2.0",
+  "basePath": "/",
+  "info": {
+    "title": "LoopBack Application",
+    "version": "1.0.0",
+  },
+  paths:{
+      ...
+  }
+}
+```
+Also note that LoopBack Next provides `validateApiSpec` in the testlab so that you can validate it in `npm test` of your app.  Isn't that cool?
+
+
+
+# LoopBack Next
+
+We're going to examine three key concepts: controller & apiSpec, components & providers, and custom sequence.  Working sample codes are available in several places, one of which is [Hello World](https://github.com/strongloop/loopback-next-hello-world).  I'm going to focus on specific conceptual roadblocks I stumbled on.
+
+Skeleton of the client code looks like this:
+```
+  const app = new Application({
+    components: [HisComponent],
+   });
+  app.controller(HelloWorldController);
+  app.sequence(MySequence);
+```
+
+### Controllers and API Spec
+
+Let's take a look at HelloWorldController code and associated API spec first, then we study what they do.
+
+```
+import {api inject} from '@loopback/core';
+import {apiSpec} from './hello-world.api';
+
+@api(apiSpec)
+export class HelloWorldController {
+  @inject('defaultName');
+  private name: string;
+  constructor() {}
+
+  helloWorld(name?: string) {
+      name = name || this.name;
+      return `Hello world ${name}!`
+  }
+}
+
+```
+```
+// hello-world.api.ts
+export const apiSpec =
+{
+  "paths": {
+    "/helloworld": {
+      "get": {
+        "x-operation-name": "helloWorld",
+        "parameters": [
+          {
+            "name": "name",
+            "in": "query",
+            "description": "Your name.",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Returns a hello world with your (optional) name."
+          }
+        }
+      }
+    }
+  }
+}
+```
+@api is a [class decorator](https://www.typescriptlang.org/docs/handbook/decorators.html).  The class decorator,
+`api` is a function that is called when the class, HelloWorldController is instantiated.  In this case, `api` registers
```

Comment:
The decorator is a function that is called when the class is defined (only once), not every time it's instantiated.